cmd/fluxctl/root_cmd.go

```diff
@@ -74,11 +82,38 @@ func (opts *rootOpts) Command() *cobra.Command {
 }
 
 func (opts *rootOpts) PersistentPreRunE(cmd *cobra.Command, _ []string) error {
+	opts.Namespace = getFromEnvIfNotSet(cmd.Flags(), "k8s-forward-namespace", opts.Namespace, envVariableNamespace)
+	opts.Token = getFromEnvIfNotSet(cmd.Flags(), "token", opts.Token, envVariableToken, envVariableCloudToken)
 	opts.URL = getFromEnvIfNotSet(cmd.Flags(), "url", opts.URL, envVariableURL)
+
+	if opts.Token != "" {
```

Comment:
We use `fluxctl` against development environments of Weave Cloud, and for this we need to be able to set both the URL and the auth token. I think it should be (in pseudo-code): if either URL or token are set, use the URL and token arguments (with defaults as they are now); otherwise, use port forwarding.

| use | set | outcome |
|-------|-----|--------------|
| OSS install | - | uses port forwarding to default namespace |
| OSS install | k8s-forward-namespace=foo | port forward to namespace "foo" |
| weave cloud | token=abc | uses default URL (https://cloud.weave.works/api/flux) and token |
| weave cloud dev | token=abc, url=http://$(minikube ip):$port/api/flux | uses URL and token |

Reply:
Good catch, fixed!